### ISSUE TITLE
Add support for JWT token authentication

### ISSUE THREAD

**Issue by mjordan** [mjordan/islandora_bagger#25]
As described in https://github.com/Islandora-CLAW/CLAW/issues/1224#issuecomment-513255325.

**Comment by alxp** [comment]
Hi Mark,

I noticed that as-is, with password authentication, islandora_bagger does not fully work with unpublished content. A few calls to Guzzle do not add the basic auth credentials, and I had to alter Drupal's routing table to allow Flysystem paths to be authenticated via basi auth.  After all that, getting the node in JSON format was denied due to a cacheable access checker, which means basic auth is not very workable.

I have gotten JWT-based authentication to work by passing the token as a parameter from Drupal.

A couple of questions:

Do you need the basic auth feature at all? If it can be completely replaced that simplifies the code, and it doesn't seem to have been working anyway.

One place where this might be a problem is if islandora_bagger is run by itself and not invoked by Drupal. in that case there'd  need to be a step where a key is retrieved.

Let me know what you think, I'll put up a PR shortly.

 - alexander

**Comment by mjordan** [comment]
> Do you need the basic auth feature at all? If it can be completely replaced that simplifies the code, and it doesn't seem to have been working anyway.

No, I only used Basic since it was the simplest (at least to me) auth method. 

> One place where this might be a problem is if islandora_bagger is run by itself and not invoked by Drupal. in that case there'd need to be a step where a key is retrieved.

I had envisioned that this would be the more common usage. For example, SFU doesn't plan on offering end users the option to generate/download a Bag, rather, Bagger will be used in a scheduled process to generate Bags for new content, which are then pushed through a preservation pipeline. The only reason we'd install the Integration Module would be for its Drupal-side logging.

I'd love a PR if you have time, but would you also be able to add the key retrieval step?

**Comment by alxp** [comment]
Looks like this is the preferred method for retrieving a JWT token after RESTful login: https://www.drupal.org/project/getjwtonlogin


Should be pretty straight forward to get this working. Will start on that now.

**Comment by alxp** [comment]
PRs ready for review / testing. No new configuration settings are required. See the "To test" sections in each PR.
